static/app/views/replays/detail/ai/useFetchReplaySummary.tsx

```diff
@@ -1,43 +1,178 @@
+  // summary data query and 2) we have the stale version of the summary data. The consuming
+  // component will briefly show a completed state before the summary data query updates.
+  const startSummaryRequestTime = useRef<number>(0);
+
+  const segmentCount = replayRecord?.count_segments ?? 0;
+
+  const {
+    mutate: startSummaryRequestMutate,
+    isError: isStartSummaryRequestError,
+    isPending: isStartSummaryRequestPending,
+  } = useMutation({
+    mutationFn: () =>
+      api.requestPromise(
+        `/projects/${organization.slug}/${project?.slug}/replays/${replayRecord?.id}/summarize/`,
+        {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          data: {
+            num_segments: segmentCount,
+          },
+        }
+      ),
+    onSuccess: () => {
+      // invalidate the query when a summary task is requested
+      // so the cached data is marked as stale.
+      queryClient.invalidateQueries({
+        queryKey: createAISummaryQueryKey(
+          organization.slug,
+          project?.slug,
+          replayRecord?.id ?? ''
+        ),
+      });
+      startSummaryRequestTime.current = Date.now();
+    },
+  });
+
+  const {
+    data: summaryData,
+    isPending,
+    isError,
+    dataUpdatedAt,
+  } = useApiQuery<SummaryResponse>(
     createAISummaryQueryKey(organization.slug, project?.slug, replayRecord?.id ?? ''),
     {
       staleTime: 0,
+      retry: false,
+      refetchInterval: query => {
+        if (isPolling(query.state.data?.[0], isStartSummaryRequestPending)) {
+          return POLL_INTERVAL_MS;
+        }
+        return false;
+      },
+      refetchOnWindowFocus: 'always',
       ...options,
     }
   );
+
+  const startSummaryRequest = useCallback(() => {
+    // Don't trigger if the feature is disabled
+    if (options?.enabled === false) {
+      return;
+    }
+
+    startSummaryRequestMutate();
+  }, [options?.enabled, startSummaryRequestMutate]);
+
+  return {
+    summaryData,
+    isPolling: isPolling(summaryData, isStartSummaryRequestPending),
+    isPending:
+      dataUpdatedAt < startSummaryRequestTime.current ||
+      isPending ||
+      summaryData?.status === ReplaySummaryStatus.PROCESSING ||
+      isStartSummaryRequestPending,
+    isError:
+      isError ||
+      summaryData?.status === ReplaySummaryStatus.ERROR ||
+      isStartSummaryRequestError,
+    startSummaryRequest,
+    isStartSummaryRequestPending,
+  };
 }
+
+const isPolling = (
+  summaryData: SummaryResponse | undefined,
+  isStartSummaryRequestPending: boolean
+) => {
+  if (!summaryData) {
+    // No data yet - poll if we've started a request
+    return isStartSummaryRequestPending;
+  }
+
+  switch (summaryData.status) {
+    case ReplaySummaryStatus.NOT_STARTED:
+      // Not started - poll if we've initiated a run
+      return isStartSummaryRequestPending;
```

Comment:
if summary has not been generated yet and...

- start summary request is pending: it means we're kind of in a waiting state, so I think "polling = true" makes sense, we don't necessarily need to wait for start summary request to finish, but if we wanted to optimize a bit (e.g. until status is not NOT_STARTED), we'd wait until not is pending

- start summary request is not pending (meaning request is completed). this should be true since we want to poll for when status is completed


based on the above, I think we either make this always return `true` if status is NOT_STARTED to simplify things _or_ return `!isStartSummaryRequestPending` for a slightly more optimized logic (but more complex to understand).